src/main/java/digital/vix/vertxdemo/controller/EndpointMetaDataController.java

```diff
@@ -0,0 +1,107 @@
+package digital.vix.vertxdemo.controller;
+
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import digital.vix.vertxdemo.dto.EndpointMetaDataDto;
+import digital.vix.vertxdemo.models.Endpoint;
+import digital.vix.vertxdemo.models.Information;
+import digital.vix.vertxdemo.service.EndpointService;
+import digital.vix.vertxdemo.service.InformationService;
+import digital.vix.vertxdemo.utils.ArrayUtils;
+import io.vertx.reactivex.core.AbstractVerticle;
+import io.vertx.reactivex.ext.web.Router;
+import io.vertx.reactivex.ext.web.RoutingContext;
+
+public class EndpointMetaDataController extends AbstractVerticle {
+
+	private Router router;
+	private ObjectMapper mapper;
+	private EndpointService endpointService;
+	private InformationService informationService;
+
+	public EndpointMetaDataController(Router router, ObjectMapper mapper, EndpointService endpointService,
+			InformationService informationService) {
+		super();
+		this.router = router;
+		this.mapper = mapper;
+		this.endpointService = endpointService;
+		this.informationService = informationService;
+	}
+
+	@Override
+	public void start() {
+		router.get("/api/endpoint").handler(this::getEndpoint);
+	}
+
+	/**
+	 * Method to determine whether multiple ids have been entered or a single id.
+	 * <br>
+	 * <br>
+	 * 
+	 * If using a single id the following should be used /api/endpoint?id=3 <br>
+	 * Calls: {@link #getEndpointById(RoutingContext)} <br>
+	 * <br>
+	 * 
+	 * Whereas when using multiple ids: /api/endpoint?ids=3%2C4$2C5 (which
+	 * translates to 3,4,5)<br>
+	 * Calls: {@link #getEndpointsByIds(RoutingContext)}<br>
+	 * 
+	 * @param routingContext
+	 */
+	public void getEndpoint(RoutingContext routingContext) {
+		if (routingContext.request().getParam("id") != null) {
+			getEndpointById(routingContext);
+		} else if (routingContext.request().getParam("ids") != null) {
+			getEndpointsByIds(routingContext);
+		}
+	}
+
+	public void getEndpointById(RoutingContext routingContext) {
+		try {
+			String idText = routingContext.request().getParam("id");
+			long id = Long.parseLong(idText);
+			endpointService.readAllEndpoints().map(jsonObject -> {
+				return mapper.readValue(jsonObject.toString(), Endpoint.class);
+			}).filter(endpoint -> endpoint.getId() == id).singleOrError().flatMap(endpoint -> {
+				return informationService.findEndpointById(id).map(information -> {
+					return new EndpointMetaDataDto(endpoint, information);
```

Comment:
I am not too experienced here but instead of using flapMap you could use just Map (without the singleOrError)?

}).filter(endpoint -> arrayContains(ids, endpoint.getId() )).map(endpoint -> {

Reply:
if it maps then the single (from the endpoint) will return a single (from the information service)

This means that we would end up with a Single<Single<EndpointMetaDataDto>>. By using a flatMap the inner single is resolved to just the EndpointMetaDataDto object, and return value is a simple Single<EndpointMetaDataDto>